### PR TITLE
Fix blendshape error (merge from 1.2)

### PR DESCRIPTION
**Pull request by luzhuang**
link: https://github.com/galacean/engine/pull/2031   https://github.com/galacean/engine/pull/2043

### CODE REVIEW

File: packages/core/src/asset/ReferResource.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export abstract class ReferResource extends EngineObject implements IReferable {
   isGCIgnored: boolean = false;
 
   private _refCount: number = 0;
+  private _superResources: ReferResource[] = null;
```
Comment (GuoLei1990) [inline packages/core/src/asset/ReferResource.ts:13]: Delete

File: packages/core/src/asset/ReferResource.ts [hunk 2]
```diff
@@ -25,17 +26,56 @@ export abstract class ReferResource extends EngineObject implements IReferable {
...
    * Destroy self.
-   * @param force - Whether to force the destruction, if it is false, refCount = 0 can be released successfully.
-   * @returns Whether the release was successful.
+   * @param force - Whether to force the destruction, if it is false, refCount = 0 can be released successfully
+   * @returns Whether the release was successful
    */
-  override destroy(force: boolean = false): boolean {
-    if (!force && this._refCount !== 0) {
-      return false;
+  override destroy(force?: boolean): boolean;
```
Comment (GuoLei1990) [inline packages/core/src/asset/ReferResource.ts:32]: Revert

File: packages/loader/src/gltf/extensions/KHR_draco_mesh_compression.ts [hunk 3]
```diff
@@ -52,8 +53,16 @@ class KHR_draco_mesh_compression extends GLTFExtensionParser {
       const buffer = GLTFUtils.getBufferViewData(bufferViews[bufferViewIndex], buffers);
       return KHR_draco_mesh_compression._decoder.decode(buffer, taskConfig).then((decodedGeometry) => {
         const mesh = new ModelMesh(engine, glTFMesh.name);
+        // @ts-ignore
+        mesh._associationSuperResource(glTFResource);
+        const meshRestoreInfo = new ModelMeshRestoreInfo();
+        meshRestoreInfo.mesh = mesh;
+        context.contentRestorer.meshes.push(meshRestoreInfo);
```
Comment (GuoLei1990) [inline packages/loader/src/gltf/extensions/KHR_draco_mesh_compression.ts:60]: Maybe this not need?

File: packages/core/src/asset/ReferResource.ts (hunk 1 above)
Comment (luzhuang) [reply packages/core/src/asset/ReferResource.ts:13]: done

File: packages/core/src/asset/ReferResource.ts (hunk 2 above)
Comment (luzhuang) [reply packages/core/src/asset/ReferResource.ts:32]: done

File: packages/loader/src/gltf/extensions/KHR_draco_mesh_compression.ts (hunk 3 above)
Comment (luzhuang) [reply packages/loader/src/gltf/extensions/KHR_draco_mesh_compression.ts:58]: GLTFMeshParser._createBlendShape need  meshRestoreInfo

**Pull request merged**